### PR TITLE
Update UG and DG for undo command

### PR DESCRIPTION
**Pull request by charlesliyifeng**
image

image

image

image

image

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -158,47 +159,40 @@ Classes used by multiple components are in the `seedu.address.commons` package.
...
 
-* `VersionedAddressBook#commit()` — Saves the current address book state in its history.
-* `VersionedAddressBook#undo()` — Restores the previous address book state from its history.
-* `VersionedAddressBook#redo()` — Restores a previously undone address book state from its history.
+The `Model` component stores a history of executed concrete commands in a stack.
+When a command is executed successfully, the command is pushed onto the stack.
+When the user executes the `undo` command, the `Model` component pops the last command from the stack and calls the `undo()` method of the command.
 
-These operations are exposed in the `Model` interface as `Model#commitAddressBook()`, `Model#undoAddressBook()` and `Model#redoAddressBook()` respectively.
+These operations are exposed in the `Model` interface as `Model#pushToUndoStack()`, `Model#undoAddressBook()`.
```
Comment (EdaYeo) [inline docs/DeveloperGuide.md:175]: Would it be better if its `Model#pushToUndoStack()` and `Model#undoAddressBook()`?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -240,11 +222,11 @@ The following activity diagram summarizes what happens when a user executes a ne
 
 **Aspect: How undo & redo executes:**
```
Comment (EdaYeo) [inline docs/DeveloperGuide.md:223]: Should redo be deleted here as well?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -240,11 +222,11 @@ The following activity diagram summarizes what happens when a user executes a ne
...
 
-* **Alternative 1 (current choice):** Saves the entire address book.
+* **Alternative 1:** Saves the entire address book.
   * Pros: Easy to implement.
   * Cons: May have performance issues in terms of memory usage.
 
-* **Alternative 2:** Individual command knows how to undo/redo by
+* **Alternative 2 (current implementation):** Individual command knows how to undo/redo by
```
Comment (EdaYeo) [inline docs/DeveloperGuide.md:229]: Should /redo be deleted here too?

**Review by EdaYeo — COMMENTED** [review]
LGTM! Just a few very minor nits.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (charlesliyifeng) [reply docs/DeveloperGuide.md:175]: It is `Model#pushToUndoStack()` and `Model#undoAddressBook()`.

**Review by EdaYeo — APPROVED** [review]

**Pull request merged**